subprojects/dependency-management/src/main/java/org/gradle/internal/component/model/GraphVariantSelector.java

```diff
@@ -46,29 +46,34 @@
 import java.util.Set;
 
 /**
- * Uses attribute matching to select a list of variants from a graph.
+ * Uses attribute matching to select a list one or more variants for a component in a graph.
```

Comment:
```suggestion
 * Uses attribute matching to select a single variant of a component in a graph.
```

The wording here seems odd. WDYT about this?

Reply:
I mean, the `GraphVariantSelectionResult` result type is literally a wrapper around a list of one or more variants (usually one, but not always).  

I think this is an odd type, and should be split into subtypes handling a single variant list and multi-variant list.  I prefer not to change this doc yet, and continue to lampshade this somewhat unexpected structure.

Reply:
Though in practice, I don't think this class ever returns an instance of `GraphVariantSelectionResult` that has more than one result. If there are multiple matching variants, it will throw an exception